src/compiler/parser.ts

```diff
@@ -5150,9 +5264,14 @@ namespace ts {
 
         // DECLARATIONS
 
+        function parseOmittedExpression() {
+            return finishNode(<OmittedExpression>createNode(SyntaxKind.OmittedExpression));
+        }
+
         function parseArrayBindingElement(): ArrayBindingElement {
             if (token() === SyntaxKind.CommaToken) {
-                return <OmittedExpression>createNode(SyntaxKind.OmittedExpression);
+                // Omitted expressions are considered expression parts, even when they do not appear in expressions - there is no great reason for this

```

Comment:
So what happens if you remove the call to `doInExpressionContext`?

Reply:
We stop reporting ` : undefined` on omitted expressions in binding patterns in out `types` baselines, mostly. I don't see any behavior changes beyond that, so it may be desirable to change.